test_users.py

```diff
@@ -0,0 +1,103 @@
+    def setUp(self):
+        """ Method called to prepare the test fixture """
+        self.newUser = Users()
+
+    def test_member_register(self):
+        """ Test to test function member_register """
+        self.newUser.users = {}
+        #users dictionary to check whether the user already exists
+        self.newUser.member_register("waweru@gmail.com", "mwaz", "password", "password")
+        result = self.newUser.member_register("waweru@gmail.com", "mwaz", "password", "password")
+        self.assertIn("205,Password Regex mismatch", result, "205,Password Regex mismatch")
+
+    def test_null_email_field(self):
+        """Test to check if email is null"""
+        result = self.newUser.member_register("", "mwaz", "password", "password")
+        self.assertIn("205,Invalid input", result, "205,Invalid input")
+
+    def test_invalid_member_email(self):
+        """Test to ceck emial pattern of member"""
+        result = self.newUser.member_register("waweru.waweru.com", "mwaz", "password", "password")
+        self.assertIn("205,Email Regex mismatch", result, "205,Email Regex mismatch")
+
+    def test_null_member(self):
+        """Test to check if member name is null"""
+        result = self.newUser.member_register("waweru@gmail.com", "", "password", "password")
+        self.assertIn("205,Regex mismatch", result, "205,Regex mismatch")
+
+    def test_null_password(self):
+        """Test to check if email is null"""
+        result = self.newUser.member_register("waweru@gmail.com", "mwaz", "", "password")
+        self.assertIn("205,Invalid input", result, "205,Invalid input")
+
+    def test_null_cpassword(self):
+        """ Test to check if cpasword is null """
+        result = self.newUser.member_register("waweru@gmail.com", "mwaz", "password", "")
+        self.assertIn("205,Invalid input", result, "205,Invalid input")
+
+    def test_cpassword_is_equal_to_password(self):
+        """Test to check whether the confirm password and the passwords are similar """
+        result = self.newUser.member_register("waweru@gmail.com", "mwaz", "password", "pass")
+        self.assertIn("400,Passwords dont match", result, "400,Passwords dont match")
+
+    def test_cpassword_is_empty(self):
+        """ Test to check whether confirm password field is empty"""
+        result = self.newUser.member_register("waweru@gmail.com", "mwaz", "password", " ")
+        self.assertIn("205,Invalid input", result, "205,Invalid input")
+
+    def test_invalid_member_name(self):
+        """Test to check if member_name follows the one specified by regex"""
+        member_registration = self.newUser.member_register("waweru@gmail.com", "&#*", "password", "password")
+        self.assertIn("205,Regex mismatch", member_registration, "205,Regex mismatch")
+
+    def test_success_login(self):
+        """ method to test a successful login """
+        self.newUser.users = {}
+        self.newUser.member_register('waweru@gmail.com', 'mwaz', 'pass', 'pass')
```

Comment:
This is used 3 times in your code. DRY! Should be in the `setUp()` method to avoid the repetition